test/token/otcEscrow.spec.ts

```diff
@@ -0,0 +1,374 @@
+
+      const vesting = Vesting__factory.connect(vestingAddress, getProvider());
+
+      expect(await vesting.index()).to.eq(index.address);
+      expect(await vesting.recipient()).to.eq(investor.address);
+      expect(await vesting.vestingAmount()).to.eq(subjectIndexAmount);
+      expect(await vesting.vestingBegin()).to.eq(subjectVestingStart);
+      expect(await vesting.vestingCliff()).to.eq(subjectVestingCliff);
+      expect(await vesting.vestingEnd()).to.eq(subjectVestingEnd);
+    });
+
+    context("when an inadequate amount of INDEX is sent to the escrow", async () => {
+
+      beforeEach(async () => {
+        subjectOtcEscrow = await deployer.token.deployOtcEscrow(
+          investor.address,
+          indexGov.address,
+          subjectVestingStart,
+          subjectVestingCliff,
+          subjectVestingEnd,
+          subjectUSDCAmount,
+          subjectIndexAmount,
+          usdc.address,
+          index.address
+        );
+        
+        await index.connect(indexGov.wallet).transfer(subjectOtcEscrow.address, ether(50));
+        await usdc.connect(investor.wallet).approve(subjectOtcEscrow.address, subjectUSDCAmount);
+      });
+
+      it("should revert", async () => {
+        await expect(subject()).to.be.revertedWith("insufficient INDEX");
+      });
+    });
+
+    context("when the sender does not have enough USDC approved", async () => {
+
+      beforeEach(async () => {
+        await usdc.connect(investor.wallet).approve(subjectOtcEscrow.address, 0);
+      });
+
+      it("should revert", async () => {
+        await expect(subject()).to.be.reverted;
+      });
+    });
+
+    context("when the sender does not have enough USDC", async () => {
+ 
+      beforeEach(async () => {
+        
+        subjectOtcEscrow = await deployer.token.deployOtcEscrow(
+          await getRandomAddress(),
+          indexGov.address,
+          subjectVestingStart,
+          subjectVestingCliff,
+          subjectVestingEnd,
+          subjectUSDCAmount,
+          subjectIndexAmount,
+          usdc.address,
+          index.address
+        );
+        
+        await index.connect(indexGov.wallet).transfer(subjectOtcEscrow.address, subjectIndexAmount);
+        await usdc.connect(investor.wallet).approve(subjectOtcEscrow.address, subjectUSDCAmount);
+      });
+
+      it("should revert", async () => {
+        await expect(subject()).to.be.reverted;
+      });
+    });
+
+    context("when the caller is unauthorized", async () => {
+
+      let subjectCaller: Account;
+
+      beforeEach(async () => {
+        subjectCaller = await getRandomAccount();
+      });
+
+      async function subject(): Promise<ContractTransaction> {
+        return await subjectOtcEscrow.connect(subjectCaller.wallet).swap();
+      }
+
+      it("should revert", async () => {
+        await expect(subject()).to.be.revertedWith("unauthorized");
+      });
+    });
+
+    context("when the caller is the investor", async () => {
```

Comment:
Let's move this up above the revert cases. Generally we want to end our tests with the revert cases